processing/src/main/java/org/apache/druid/segment/transform/ExpressionTransform.java

```diff
@@ -100,7 +100,11 @@ static class ExpressionRowFunction implements RowFunction
     public Object eval(final Row row)
     {
       try {
-        return expr.eval(InputBindings.forRow(row)).valueOrDefault();
+        Object result = expr.eval(InputBindings.forRow(row)).valueOrDefault();
+        if (result != null && result.getClass().isArray()) {
+          return Rows.objectToStrings(result);
+        }
+        return result;
```

Comment:
this should not change, we want ARRAY functions to output arrays, the proper place to change is  https://github.com/apache/druid/blob/master/processing/src/main/java/org/apache/druid/segment/transform/ExpressionTransform.java#L115 which no longer needs to call `ExpressionSelectors.coerceEvalToObjectOrList`

Reply:
we dont really need this change. Removed it.